### PR TITLE
docs: improve consuming partial-Ivy code outside the Angular CLI

### LINKED ISSUE #44026: Angular 13: document config changes necessary when using @ngtools/webpack
Hi,

When trying to update my angular app built using @ngtools/webpack from angular 12 to the upcoming angular 13, I get the following error at runtime:

```
JIT compilation failed for injectable 
class PlatformLocation {}
core.mjs:4060
Uncaught Error: The injectable 'PlatformLocation' needs to be compiled using the JIT compiler, but '@angular/compiler' is not available.

The injectable is part of a library that has been partially compiled.
However, the Angular Linker has not processed the library such that JIT compilation is used as fallback.

Ideally, the library is processed using the Angular Linker to become fully AOT compiled.
Alternatively, the JIT compiler should be loaded by bootstrapping using '@angular/platform-browser-dynamic' or '@angular/platform-server',
or manually provide the compiler with 'import "@angular/compiler";' before bootstrapping.
    Angular 3
    Webpack 9
core.mjs:4079
```

The same project works fine when built with angular 12.x (with no other change than a minor adjustment in webpack.config.mjs) so it looks like a regression.

I made a minimal self-contained repository to reproduce the issue at https://github.com/vanackere/ngtools-webpack-bug

To reproduce, simply run `npm run dev` then `npm run serve`

Thanks by advance for looking at this !

### PR DESCRIPTION
**Pull request by alan-agius4**
With this change we add an example of how to consume the linker plugin and also clean up the parts of this section.

Closes #44026

### CODE REVIEW

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Looks good @alan-agius4.
We are trying to avoid adding inline code snippets to the docs.
Do you need any help moving this to an external example?

**Review by petebacondarwin — APPROVED** [review]
Great work @alan-agius4 thanks.
Strictly the second commit should probably be a `build(docs-infra): ...` but won't block the merge over it.

File: aio/tools/transforms/examples-package/services/region-parser.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = function regionParser() {
   regionParserImpl.regionMatchers = {
     ts: inlineC,
     js: inlineC,
+    mjs: inlineCOnly,
```
Comment (gkalpak) [inline aio/tools/transforms/examples-package/services/region-parser.js:13]: OOC, why not `inlineC`?

Comment (alan-agius4) [reply aio/tools/transforms/examples-package/services/region-parser.js:13]: Because of the docs plaster comment, looks neater using `// …` IMHO.

**Review by atscott — APPROVED** [review]

**Comment by atscott** [comment]
This PR was merged into the repository by commit 36388b3ab23578151d7edad5106a26af69bcb45e.

**Pull request closed without merging**